src/vmEntry.h

```diff
@@ -144,8 +143,16 @@ class VM {
         return _zing;
     }
 
-    static bool canSampleObjects() {
-        return _can_sample_objects;
+    static bool addCanSampleObjectsCapability() {
+        jvmtiCapabilities capabilities = {0};
+        capabilities.can_generate_sampled_object_alloc_events = 1;
+        return _jvmti->AddCapabilities(&capabilities) == 0;
+    }
+
+    static void relinquishCanSampleObjectsCapability() {
```

Comment:
A bit too complicated name. I know it matches JVMTI function, but I think `releaseSampleObjectsCapability` would be better. The other function will be correspondigly `addSampleObjectsCapability`.